File: src/test/java/com/indiepost/service/SitemapServiceTests.java
package com.indiepost.service;

import com.indiepost.NewIndiepostApplication;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.context.web.WebAppConfiguration;

import javax.inject.Inject;
import java.net.MalformedURLException;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

/**
 * Created by jake on 17. 3. 21.
 */

@RunWith(SpringRunner.class)
@SpringBootTest(classes = NewIndiepostApplication.class)
@WebAppConfiguration
public class SitemapServiceTests {

    @Inject
    private SitemapService sitemapService;

    @Test
    public void sitemapShouldCreateCorrectly() throws MalformedURLException {
        String sitemap = sitemapService.buildSitemap();
        assertThat(sitemap).isNotEmpty();
        System.out.println(sitemap);
    }
}

File: src/test/kotlin/com/indiepost/service/SitemapServiceTests.kt
package com.indiepost.service

import com.indiepost.IndiepostBackendApplication
import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.test.context.web.WebAppConfiguration
import javax.inject.Inject

/**
 * Created by jake on 17. 3. 21.
 */

@ExtendWith(SpringExtension::class)
@SpringBootTest(classes = arrayOf(IndiepostBackendApplication::class))
@WebAppConfiguration
class SitemapServiceTests {

    @Inject
    private lateinit var sitemapService: SitemapService

    @Test
    fun sitemapShouldCreateCorrectly() {
        val sitemap = sitemapService.buildSitemap()
        assertThat(sitemap).isNotEmpty()
        println(sitemap)
    }
}
